app/models/user.js

```diff
@@ -195,7 +157,14 @@ exports.addModel = function(database) {
       that.id = uuid.v4()
 
       that.validateOnCreate()
-        .then(function(user) { return user.updateHashedPassword() })
+        .then(function(user) {
+          if (_.isFunction(user.hashedPassword)) {
+            var pwd = user.hashedPassword()
+            return user.updatePassword(pwd, pwd)
+          } else {
+            resolve(user)
```

Comment:
I agree there are static methods Promise.resolve, Promise.rejects and others, though this particular line does not use them, i.e. it uses function that is defined in our main Promise https://github.com/pepyatka/pepyatka-server/pull/121#discussion-diff-28286455L190. How does it work then? It just does not execute that line at all (you can run `jake coverage` to ensure). And that totally makes sense -- we call `create` method only for a user with a plain-text password.

Let's take a step back and look at constructor:
```
    if (!_.isUndefined(params.hashedPassword)) {
      this.hashedPassword = params.hashedPassword
    } else if (!_.isUndefined(params.password)) {
      this.hashedPassword = function() { return params.password; }  // just a hack, for temporary storing unhashed password
    }
```
we create a plain-text password and later call `var pwd = user.hashedPassword()` method, however we *already* know that this user is going to have a plain text one.

Taking all this into account I would suggest restoring original constructor:
```
    this.hashedPassword = params.hashedPassword
    this.password = params.password
```
and replacing this `.then` to just
```
.then(function(user) { user.updatePassword(user.password, user.password) }
```

Note, that we never store or set plain-text password for users that are already in our storage, they have just hash.